templates/web-application/webpack.dev.js

```diff
@@ -0,0 +1,89 @@
+// @ts-check
+import path from "node:path";
+import { fileURLToPath } from "node:url";
+import HtmlWebpackPlugin from "html-webpack-plugin";
+import ReactRefreshWebpackPlugin from "@pmmmwh/react-refresh-webpack-plugin";
+import open from "open";
+
+import { loadSwcConfig } from "./loadSwcConfig.js";
+
+/** @type {import("webpack").Configuration} */
+
+class OpenBrowser {
+    isFirstCompile = true;
+
+    apply(compiler) {
+        compiler.hooks.done.tap("done", async () => {
+            if (this.isFirstCompile) {
+                console.log("\nOpening browser at: \"http://localhost:8080/\"");
+
+                await open("http://localhost:8080/");
+            }
+            this.isFirstCompile = false;
+        });
+    }
+}
+
+export default {
+    mode: "development",
+    target: "web",
+    devtool: "eval-cheap-module-source-map",
```

Comment:
Quick dev optimisation, see:
- https://webpack.js.org/guides/build-performance/#devtool
- https://webpack.js.org/configuration/devtool/#devtool

Reply:
Might be worth to actually add those comments in the config so future maintainers understand why it's configured this way.